app.py

```diff
@@ -42,20 +37,29 @@ def ocrmypdf_api():
     if uploaded_file.filename == "":
         return "No file selected", 400
 
-    if uploaded_file.mimetype != "application/pdf":
-        return "Invalid file type, please upload a PDF file", 400
+    allowed_extensions = {".pdf", ".jpg", ".jpeg", ".png"}
+    file_extension = os.path.splitext(uploaded_file.filename)[1]
+
+    if file_extension not in allowed_extensions:
+        return "Invalid file type, please upload a PDF or image file", 400
+
+    if file_extension in {".jpg", ".jpeg", ".png"}:
+        # convert the image to PDF using ocrmypdf's native image handling
+        with BytesIO(uploaded_file.read()) as image_buffer, BytesIO() as pdf_buffer:
+            ocrmypdf.ocr(image_buffer, pdf_buffer, image_dpi=300)
+            pdf_buffer.seek(0)
+            pdf_file = pdf_buffer.read()
+    else:
+        pdf_file = uploaded_file.read()
 
-    filename_parts = os.path.splitext(uploaded_file.filename)
     ocr_options = request.form.get("ocr_options")
 
     # Read the PDF file and perform OCR on it
-    with NamedTemporaryFile(suffix=filename_parts[1]) as output_file, BytesIO(
-        uploaded_file.read()
-    ) as input_buffer:
+    with BytesIO(pdf_file) as pdf_output_file, BytesIO() as output_file:
```

Comment:
there's a reason for one of the tempfile usage here to workaround RAM limit when the running instance is highly constraint on that resource